### PR TITLE
feat: TEC-578, add 'More' dropdown

### PR DESCRIPTION
**Pull request by gcedo**
live @ http://gcedo.github.io/component-navigation/ (some icons return a 404 on `gh-pages`, you may want to clone it locally)

BREAKING CHANGE

- The following dependencies were updated:
    - component-sections-card, needed support for custom sections titles
    - component-accordion, needed to display a horizontal rule
    - component-icon, needed the radio icon
- The 'More' navbar link is now a Balloon component, which displays a SectionsCard component with links divided into three sections:
    - Apps and digital editions
    - Media
    - From the economist group
- The Media section now features also 'Radio' and 'Apps'
- The mobile menu was changed as follows:
    - Print Edition is moved right before the Media icons
    - Apps and Radio are added to the Media icons
    - Products is added right after the Media icons, displaying the links from the More SectionsCard

### CODE REVIEW

**Comment by fabiosantoscode** [comment]
:rainbow: for me. Looking good!

File: src/index.css [hunk 1]
```diff
@@ -22,12 +24,29 @@
...
+  display: inline-block;
+  height: 100%;
+}
+/* stylelint-enable */
+
+.navigation__more .list {
+  column-count: 1;
+}
+
+.navigation__more .sections-card__list-sections {
```
Comment (umbobabo) [inline src/index.css]: L46 can been unified with this. The selectors are now not anymore corrected because they refer to the previous implementation but we can keep as they are if you have to refactor sections card for this.

File: src/index.js [hunk 2]
```diff
@@ -3,10 +3,12 @@ import Balloon from '@economist/component-balloon';
 import Button from '@economist/component-link-button';
 import GoogleSearch from '@economist/component-google-search';
 import Icon from '@economist/component-icon';
+import More from './parts/More';
 import React from 'react';
 import SectionsCard from '@economist/component-sections-card';
 import StickyPosition from 'react-sticky-position';
 
+
```
Comment (umbobabo) [inline src/index.js]: useless empty line

File: src/example.js [hunk 3]
```diff
@@ -30,22 +29,148 @@ const sharedMenu = {
...
+      'href': 'https://gre.economist.com/',
+    },
+    {
+      'title': 'Learning.ly',
+      'href': 'https://learning.ly',
+      internal: false,
+    },
+    {
+      'title': '1843',
+      'href': '/help/about-us',
```
Comment (umbobabo) [inline src/example.js]: Are we sure this is a relative link?

File: src/example.js [hunk 4]
```diff
@@ -30,22 +29,148 @@ const sharedMenu = {
...
+      internal: false,
+    },
+    {
+      'title': 'Economist on iPad',
+      'href': 'https://search.itunes.apple.com/WebObjects/MZContentLink.woa/wa/link?path=apps%2ftheeconomist',
+      internal: false,
+    },
+    {
+      'title': 'Espresso',
+      'href': '/digital',
```
Comment (umbobabo) [inline src/example.js]: Are we sure this is correct?

File: src/example.js [hunk 5]
```diff
@@ -30,22 +29,148 @@ const sharedMenu = {
...
+      'href': 'http://films.economist.com/',
+      'internal': false,
+      'icon': {
+        'useBackground': true,
+        'color': 'chicago',
+        'icon': 'film',
+      },
+    },
+  ],
+  'blogs': [
```
Comment (umbobabo) [inline src/example.js]: Are these blogs?

File: src/example.js [hunk 6]
```diff
@@ -30,22 +29,148 @@ const sharedMenu = {
...
   {
     title: 'Print Edition',
     href: '/printedition',
   },
+  ...navigationLinks.media,
+  {
+    title: 'Products',
+    children: [
+      ...moreBalloonData.sections,
+      { hr: true }, // eslint-disable-line id-length
```
Comment (umbobabo) [inline src/example.js]: Do we need an HR to draw a row?
This is not really semantically correct on a list as we are using. I would say that we should use a class applied to the element that should create the visual break.

File: src/index.css [hunk 7]
```diff
@@ -12,6 +12,8 @@
 @import '@economist/component-icon/backgrounds/magnifier.css';
 @import '@economist/component-icon/backgrounds/user.css';
 @import '@economist/component-icon/backgrounds/close.css';
+@import '@economist/component-icon/backgrounds/radio.css';
+@import '@economist/component-icon/backgrounds/apps.css';
```
Comment (umbobabo) [inline src/index.css]: Alphabetic order is preferable in this case.

File: src/index.css [hunk 8]
```diff
@@ -22,12 +24,29 @@
 /* stylelint-disable */
 .navigation .navigation__main-navigation-link,
 .navigation .navigation__main-navigation-link > a {
-/* stylelint-enable */
   box-sizing: border-box;
   height: var(--navigation__default-height);
   text-decoration: none;
 }
 
+.navigation .navigation__main-navigation-link > span {
```
Comment (umbobabo) [inline src/index.css]: I know that some error were already there, anyway let try to avoid the direct use of the DOM when doable.

File: src/parts/more.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import React, { PropTypes as T } from 'react';
```
Comment (umbobabo) [inline src/parts/more.js]: If you create a part for the more I think You should do the same for the topic menu.

File: test/index.js [hunk 10]
```diff
@@ -6,7 +6,7 @@ import chaiReactElement from 'chai-react-element';
 import links from './links';
 chai.use(chaiReactElement).should();
 // Get data.
-import navgiationLinks from '@economist/component-sections-card/context';
+import navgiationLinks from '@economist/component-sections-card/lib/context';
```
Comment (umbobabo) [inline test/index.js]: Can we fix this typo?

File: src/example.js (hunk 3 above)
Comment (fabiosantoscode) [inline src/example.js]: yup!

Comment (fabiosantoscode) [inline src/example.js]: We have been using relative links pretty much everywhere. The downside is that they don't work on stage but quite honestly I don't think we're even supposed to care.

File: src/parts/more.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+import Balloon from '@economist/component-balloon';
+import SectionsCard from '@economist/component-sections-card';
+
+function More({ href, title, moreBalloonData }) {
```
Comment (umbobabo) [inline src/parts/more.js]: More, especially for the file name, is really generic and barely meaningful.
Due to the nature of the element we could use or a folder /menu or a name like "menu-more.js", maybe?

File: src/index.js (hunk 2 above)
Comment (gcedo) [inline src/index.js]: :+1:

File: src/example.js (hunk 4 above)
Comment (gcedo) [inline src/example.js]: In the story there's the google doc to double check them https://jira.economist.com/browse/TEC-578

File: src/example.js (hunk 5 above)
Comment (gcedo) [inline src/example.js]: No, but the `props` of `component-sections-card` are 
- sections
- media
- blogs
 that refer to the first, second and third column respectively. These are not generic names and should be changed, but I don't want to break these component contract in the context of this story

File: src/example.js (hunk 6 above)
Comment (gcedo) [inline src/example.js]: This is out of scope for this PR. The `hr` belongs to `component-accordion` https://github.com/economist-components/component-accordion/pull/17.

File: src/index.css (hunk 7 above)
Comment (gcedo) [inline src/index.css]: :+1:

File: test/index.js (hunk 10 above)
Comment (gcedo) [inline test/index.js]: :+1:

File: src/parts/more.js (hunk 11 above)
Comment (gcedo) [inline src/parts/more.js]: :+1:
